### ISSUE TITLE
Rename methods in TestTemplateInvocationContextProvider

### ISSUE THREAD

**Issue by sbrannen** [junit-team/junit5#766]
## Status Quo

The names of methods in the `TestTemplateInvocationContextProvider` API introduced in M4 are too generic.

Consequently, there is a good chance that we will later run into naming collisions with other extension APIs that _support_ or _provide_ for a given container context. Such collisions would effectively prevent a single extension from being able to implement two such affected extension APIs. For that reason, this issue is marked as a _bug_, since it's essentially a bug waiting to happen.

## Related Issues

- This issue depends on #793 

## Deliverables

- [ ] Rename the `supports()` and `provide()` methods in `TestTemplateInvocationContextProvider` so that naming collisions can be avoided -- akin to how the sole method in `TestInstancePostProcessor` is named `postProcessTestInstance()` as opposed to simply `postProcess()`.

**Comment by Andrei94** [comment]
@sbrannen on it 😄

**Comment by sbrannen** [comment]
@Andrei94, this is more of an "internal task" for the core team; however, if you would like to make proposals for the new method names, feel free to do so.

**Comment by Andrei94** [comment]
how about `supportsExecutionContext()` and `providesExecutionContext()`?

**Comment by sbrannen** [comment]
> how about `supportsExecutionContext()` and `providesExecutionContext()`?

That's unfortunately just as ambiguous as the status quo since the method signatures already reference the execution context as method parameters.

To keep matters simple, I am going to assign this task to myself, which I probably should have done in the first place.

Rationale: I already know the direction we want to go, and I will have to review any proposals in detail anyway. So, in the end, it will go faster if I just tackle this one.

Cheers!

**Comment by sbrannen** [comment]
@Andrei94, you might be interested in checking out issues that are marked as [up for grabs](https://github.com/junit-team/junit5/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs).

Otherwise, if an issue is not currently assigned to somebody, feel free to ask if you can contribute.

**Comment by Andrei94** [comment]
@sbrannen, ok. I will check those as well.

Thank you for the hints.

**Comment by sbrannen** [comment]
Note that this issue now depends on #793.

**Comment by sbrannen** [comment]
Closing this issue since it has been superseded by #793.
